### DISCUSSION TITLE
Range estimation/ way to show effeciency per km/mi

### DISCUSSION THREAD

**Discussion by WvBijsterveld** [danielnilsson9/bbs-fw#179] [Ideas]
Is there any way to get a estimate range in imperial or metric based on Wh/Km/mi or directly showing your active draw per km or 10 km's.

Suggestions for compiling my own would be welcome as well.

**Comment by EigenMania** [comment]
You could write a function that integrates the battery percentage and distance driven (based on wheel size and RPM), computes the "efficiency" (km driven / percent drop), and then estimates the range left from the current battery percentage. Then you could write this to the display during walk mode.

**Reply by WvBijsterveld** [reply]
Is Wh/km harder to do/ not really possible? I want to try and see what my rough range would be if using lower pas settings, however don't want to drain the battery alot on that pas setting since that feels a bit time consuming.

**Reply by EigenMania** [reply]
Just not sure how tricky integrating the motor current draw would be (to compute Wh) plus you cannot store remaining capacity on the firmware easily. But still maybe possible. I think you'd just need to experiment a bit and see what works.

**Reply by WvBijsterveld** [reply]
ok will do thank you!
